### PR TITLE
admin api service

### PR DESCRIPTION
**Pull request by jeffgrunewald**
split out rpcs used to manage the config service itself to group them according to function

### CODE REVIEW

File: src/service/iot_config.proto [hunk 1]
```diff
@@ -346,14 +353,25 @@ message gateway_load_region_req_v1 {
...
 
-message gateway_location_req_v1 {
-  bytes gateway = 1;
+message admin_add_key_req_v1 {
+  enum key_type_v1 {
+    operator = 0;
+    packet_router = 1;
+  }
+
+  string pubkey = 1;
```
Comment (madninja) [inline src/service/iot_config.proto:364]: So pub keys are still passed around as bytes and not as strings right?

File: src/service/iot_config.proto [hunk 2]
```diff
@@ -346,14 +353,25 @@ message gateway_load_region_req_v1 {
...
+  key_type_v1 key_type = 2;
+  bytes signature = 3;
```
Comment (madninja) [inline src/service/iot_config.proto:366]: Document this field stating that this request needs to be signed by an existing admin key (vs the pub key being added)

File: src/service/iot_config.proto [hunk 3]
```diff
@@ -346,14 +353,25 @@ message gateway_load_region_req_v1 {
...
+}
+
+message admin_remove_key_req_v1 {
+  string pubkey = 1;
```
Comment (madninja) [inline src/service/iot_config.proto:370]: Same? bytes?

File: src/service/iot_config.proto [hunk 4]
```diff
@@ -346,14 +353,25 @@ message gateway_load_region_req_v1 {
...
   bytes signature = 2;
```
Comment (madninja) [inline src/service/iot_config.proto:371]: Document who's expected to be signing

File: src/service/iot_config.proto (hunk 1 above)
Comment (jeffgrunewald) [reply src/service/iot_config.proto:364]: yeah, i was thinking of this only from the perspective of the operator using this but bytes is consistent across all the other uses of passing around a pubkey and it's easy enough to base64 encode them

Comment (jeffgrunewald) [reply src/service/iot_config.proto]: fixed [ca3da06](https://github.com/helium/proto/pull/288/commits/ca3da06d51faa7b490e2377cf47c20098e9900c5)

File: src/service/iot_config.proto (hunk 2 above)
Comment (jeffgrunewald) [reply src/service/iot_config.proto:370]: added [ca3da06](https://github.com/helium/proto/pull/288/commits/ca3da06d51faa7b490e2377cf47c20098e9900c5)

File: src/service/iot_config.proto (hunk 1 above)
Comment (madninja) [reply src/service/iot_config.proto]: the user passes it to the cli as a public key (b58 string). The cli would then decode it.. you're not base64 encoding keys now are you?!

Comment (jeffgrunewald) [reply src/service/iot_config.proto]: You’re right, b58, not 64

File: src/service/iot_config.proto [hunk 5]
```diff
@@ -346,14 +353,31 @@ message gateway_load_region_req_v1 {
...
 
-message gateway_load_region_res_v1 {}
+message admin_load_region_res_v1 {}
 
-message gateway_location_req_v1 {
-  bytes gateway = 1;
+message admin_add_key_req_v1 {
+  enum key_type_v1 {
+    // administrator key
+    operator = 0;
```
Comment (michaeldjeffrey) [inline src/service/iot_config.proto:361]: Can we call this `administrator`? 

`operator` has been used in the past to mean many things other than `admin`.

Comment (jeffgrunewald) [reply src/service/iot_config.proto:361]: sure thing

Comment (jeffgrunewald) [reply src/service/iot_config.proto]: [89b4930](https://github.com/helium/proto/pull/288/commits/89b493096d25504180a737e8304ff011a8ce50cb)

**Review by michaeldjeffrey — APPROVED** [review]

**Pull request merged**